File: data/imported/knowledge/slides/java/merged/snippets/snippet_031.java
@Test
    void testBlur() throws IOException {
        try (var pres = new Presentation()) {
            var slide = clear(pres);
            var shape = slide.getShapes().addAutoShape(ShapeType.RECTANGLE, 100, 100, 200, 100);
            var ef = shape.getEffectFormat();
            ef.setBlurEffect(8, true);

            try (var pres2 = saveAndReopen(pres)) {
                var b2 = pres2.getSlides().get(0).getShapes().get(0).getEffectFormat().getBlurEffect();
                assertThat(b2).as("blur_effect should not be None after reload").isNotNull();
                assertThat(b2.getRadius()).isEqualTo(8);
            }
        }
    }

File: data/imported/knowledge/slides/cpp/merged/snippets/snippet_031.cpp
TEST_F(ImagesIntegrationTest, PictureFrame) {
    Presentation pres;
    auto png = create_test_png(0, 0, 255);
    auto& img = pres.images().add_image(std::span<const std::uint8_t>(png));
    pres.slides()[0].shapes().add_picture_frame(
        ShapeType::RECTANGLE, 50, 50, 100, 100, img);
    EXPECT_GE(pres.slides()[0].shapes().size(), 1u);

    auto pres2 = save_and_reopen(pres);
    EXPECT_GE(pres2.slides()[0].shapes().size(), 1u);
}